packages/storage/src/persisted.ts

```diff
@@ -172,8 +172,18 @@ export function makePersisted<T, O extends Record<string, any> = {}>(
   const init = storage.getItem(name, storageOptions);
   const set =
     typeof signal[0] === "function"
-      ? (data: string) => (signal[1] as any)(() => deserialize(data))
-      : (data: string) => (signal[1] as any)(reconcile(deserialize(data)));
+      ? (data: string) => {
+        try {
+          const value = deserialize(data);
+          (signal[1] as any)(() => value);
+        } catch (e) { }
```

Comment:
Shouldn't catching errors be the job of the (de)serializer?
Instead of try catching it here, I would modify the default JSON serializer to prevent throwing.

```ts
const default_serialize: (data: any) => string = 
	data => {
		try {
			return JSON.stringify(data)
		} catch (e) {
			if (isDev) console.warn(e)
		}
	}
	
const default_deserialize: (data: string) => any = 
	data => {
		try {
			return JSON.parse(data)
		} catch (e) {
			if (isDev) console.warn(e)
		}
	}

// in makePersisted
const serialize: (data: T) => string = options.serialize || default_serialize
const deserialize: (data: string) => T = options.deserialize || default_deserialize
```

Reply:
I considered it, but this would mean you'd have to repeat the code for every serializer. For the sake of minimal bundle size, I decided against it.

Reply:
only for the serializes that can themselves throw, right?
but fine, if that's the trade-off you're choosing
but can you add some kind of dev-only warning like I did?
Right now you will have no idea that serialization failed silently.

Reply:
In case of web socket or broadcast channel or pusher, it might receive data that fails to deserialize because it is unrelated... but a warning doesn't hurt, so that's fine.

Reply:
I would prefer not having the warning, the library should only warn if it fails at decoding messages sent by itself. It would be very confusing if we keep receiving warnings that third party messages aren't decoded properly as they should not be decoded in the first place.

Reply:
Since we cannot know in advance if a message is sent by ourselves, this requirement is a bit problematic.

Reply:
Why do we need to serialize the messages to string, I have patched it this way and it worked well for me.
```JavaScript
            (subscriber) =>
              window.addEventListener("message", (ev) => {
                if (ev.data.type === "createCookieSignal") subscriber(ev.data);
              }),
            (key, newValue) => {
              postMessage({ key, newValue, type: "createCookieSignal", url: location.href });
            },
```

Reply:
If I apply the current patch as-is, I get hundreds of logs per minute

![image](https://github.com/solidjs-community/solid-primitives/assets/7070881/ca67c1fc-2237-4f85-b253-a514036a1aa0)

Those messages are caused by a bug in the React DevTools Chrome extension, but they could very well be legitimate.
https://github.com/facebook/react/issues/27992

This library can't warn on non-JSON messages.

Reply:
Ah, the original postMessage API only supported string; I see this shortcoming was fixed in a later version. Since that is the case, we can certainly drop the serialization for post message / broadcast channel sync.